### DISCUSSION TITLE
Session Expired, Logging in again.....

### DISCUSSION THREAD

**Discussion by AdvookIn** [TechfaneTechnologies/QtsApp#5] [Q&A] [answered]
```
[python F:\Market\QtsApp\QtsApp\example.py
Session Expired, Logging in again.....
2.3.67 None None .quantsapp.com Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36 Edg/103.0.1264.77
Traceback (most recent call last):
  File "F:\Market\QtsApp\QtsApp\example.py", line 13, in <module>
    QtsAppRun()
  File "F:\Market\QtsApp\QtsApp\qtsapp\QtsApp.py", line 2088, in QtsAppRun
    qtsapp = QTSAppUser(ws_name=ws_name)
  File "F:\Market\QtsApp\QtsApp\qtsapp\QtsApp.py", line 231, in __init__
    self._validate_session()
  File "F:\Market\QtsApp\QtsApp\qtsapp\QtsApp.py", line 283, in _validate_session
    self._api_key, self._access_token = self._auth()
  File "F:\Market\QtsApp\QtsApp\qtsapp\QtsApp.py", line 277, in _auth
    f.write(f"API_KEY={msg['api_key']}\n")
KeyError: 'api_key'](url)
```

**Comment by MooneDrJune** [comment, marked as answer]
> ```
> [python F:\Market\QtsApp\QtsApp\example.py
> Session Expired, Logging in again.....
> 2.3.67 None None .quantsapp.com Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36 Edg/103.0.1264.77
> Traceback (most recent call last):
>   File "F:\Market\QtsApp\QtsApp\example.py", line 13, in <module>
>     QtsAppRun()
>   File "F:\Market\QtsApp\QtsApp\qtsapp\QtsApp.py", line 2088, in QtsAppRun
>     qtsapp = QTSAppUser(ws_name=ws_name)
>   File "F:\Market\QtsApp\QtsApp\qtsapp\QtsApp.py", line 231, in __init__
>     self._validate_session()
>   File "F:\Market\QtsApp\QtsApp\qtsapp\QtsApp.py", line 283, in _validate_session
>     self._api_key, self._access_token = self._auth()
>   File "F:\Market\QtsApp\QtsApp\qtsapp\QtsApp.py", line 277, in _auth
>     f.write(f"API_KEY={msg['api_key']}\n")
> KeyError: 'api_key'](url)
> ```

It Seems You are using google sign in method, Hence Auto Updating of `API_KEY` & `ACCESS_TOKEN` wont happen, you have to do it manually. And update the same in `.env` file instead of  `.env.secret`, To know How To Do The Same, Please Follow This [Guideline](https://github.com/TechfaneTechnologies/QtsApp/blob/main/GetApiKeyAndAccessTokenFromBrowser.md)
